x-pack/plugins/fleet/server/services/epm/elasticsearch/template/install.ts

```diff
@@ -365,8 +365,8 @@ export function buildComponentTemplates(params: {
           mapping: {
             ...templateSettings.index?.mapping,
             total_fields: {
+              limit: '1000',
```

Comment:
ℹ️ Allow the template settings to override the default value. The old logic would always override the total fields with 10k, no matter what's specified in the manifest.